code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/configuration/MobileIdentitiesProvider.kt

```diff
@@ -0,0 +1,455 @@
+    private const val JSON_VALUE_NAMESPACE_MCPNS_DPID = "20919"
+
+    // JSON value constants for SDKIdentifier type
+    private const val JSON_VALUE_TYPE_ANALYTICS = "analytics"
+    private const val JSON_VALUE_TYPE_INTEGRATION_CODE = "integrationCode"
+    private const val JSON_VALUE_TYPE_TARGET = "target"
+    private const val JSON_VALUE_TYPE_NAMESPACE_ID = "namespaceId"
+
+    private val REQUIRED_READY_EXTENSIONS = listOf(
+        SharedStateKeys.Analytics.EXTENSION_NAME,
+        SharedStateKeys.Audience.EXTENSION_NAME,
+        SharedStateKeys.Configuration.EXTENSION_NAME,
+        SharedStateKeys.Target.EXTENSION_NAME,
+        SharedStateKeys.Identity.EXTENSION_NAME,
+    )
+
+    @VisibleForTesting
+    internal object SharedStateKeys {
+        internal object Analytics {
+            internal const val EXTENSION_NAME = "com.adobe.module.analytics"
+            internal const val ANALYTICS_ID = "aid"
+            internal const val VISITOR_IDENTIFIER = "vid"
+        }
+
+        internal object Audience {
+            internal const val EXTENSION_NAME = "com.adobe.module.audience"
+            internal const val DPID = "dpid"
+            internal const val DPUUID = "dpuuid"
+            internal const val UUID = "uuid"
+        }
+
+        internal object Identity {
+            internal const val EXTENSION_NAME = "com.adobe.module.identity"
+            internal const val MID = "mid"
+            internal const val VISITOR_IDS_LIST = "visitoridslist"
+            internal const val ADVERTISING_IDENTIFIER = "advertisingidentifier"
+            internal const val PUSH_IDENTIFIER = "pushidentifier"
+        }
+
+        internal object Target {
+            internal const val EXTENSION_NAME = "com.adobe.module.target"
+            internal const val TNT_ID = "tntid"
+            internal const val THIRD_PARTY_ID = "thirdpartyid"
+        }
+
+        internal object Configuration {
+            internal const val EXTENSION_NAME = "com.adobe.module.configuration"
+            internal const val CONFIG_EXPERIENCE_CLOUD_ORG_ID = "experienceCloud.org"
+        }
+    }
+
+    /**
+     * Represents an identifier known to the SDK
+     */
+    private data class ID(val namespace: String, val value: String, val type: String)
+
+    /**
+     * Determines if all the shared states required to collect identities are ready.
+     * A state is ready if [SharedStateResult.status] is not [SharedStateStatus.PENDING]
+     * The required shared states are [REQUIRED_READY_EXTENSIONS]
+     *
+     * @param event an [Event] generated by the GetSdkIdentities API
+     * @param extensionApi extensionApi
+     * @return true, if the all the shared states required are not [SharedStateStatus.PENDING];
+     *         false, if any one of the shared states required [SharedStateResult.PENDING]
+     */
+    internal fun areAllSharedStatesReady(event: Event, extensionApi: ExtensionApi): Boolean {
```

Comment:
This method will not be required because we decided to go with returning all the available identifiers at the time of query.